crates/rustc_codegen_spirv/src/codegen_cx/entry.rs

```diff
@@ -554,4 +554,38 @@ impl<'tcx> CodegenCx<'tcx> {
             }
         }
     }
+
+    fn location_size_of_type(&self, ty: Word) -> u32 {
+        match self.lookup_type(ty) {
+            // Arrays take up multiple locations.
+            SpirvType::Array { count, .. } => {
+                self.builder
+                    .lookup_const_u64(count)
+                    .expect("Array type has invalid count value") as u32
+            }
```

Comment:
What happens if there's e.g. an array of structs? Shouldn't this recurse on the element type?